ls1intum/Artemis
# Feature/programming exercise/hints

### Checklist

### Motivation and Context



Add hints to a programming exercise to help students solve it.

### Description


- Implemenation of new entity _exercise_hint_ (not specific to programmingExercises!)
- New overview/editing/detail screens for managing exercise_hints (currently only active for programmingExercises)
- Show question mark icon in student code editor if there is at least 1 hint (and code editor admin view when assignment repository is selected), opens modal with hints on click. 

Not part of this PR, but could be done in a follow up:
- Only show hints when the student has failed a couple of times
- Add hints as a markdown editor command



- [x] Generate boilerplate with jhipster for new entity
- [x] Add new entity exercise_hint
- [x] Fix boilerplate
- [x] Use markdown editor for editing hint content
- [x] Show hints in programming exercise if exist

### Steps for Testing


1. Log in to ArTEMiS
2. Go to programming-exercises > manage hints
3. Create a couple of hints
4. Open the code editor as a student
5. Click on hints and check if they are rendered correctly

### Screenshots



**Exercise Hint overview:**

![image](https://user-images.githubusercontent.com/28230611/61370922-d47afe80-a893-11e9-841b-2a2f80738e34.png)

**Exercise Hint edit:**

![image](https://user-images.githubusercontent.com/28230611/61370963-e9579200-a893-11e9-8c42-979d67272001.png)

**Student view:**

![image](https://user-images.githubusercontent.com/28230611/61391504-1f5d3c00-a8bd-11e9-9d1c-05de341c9133.png)


![image](https://user-images.githubusercontent.com/28230611/61370888-c0370180-a893-11e9-920c-b5997cc545c5.png)
